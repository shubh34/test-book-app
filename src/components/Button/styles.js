import styled from 'styled-components';
import type { ComponentType } from 'react';

export const StyledButton: ComponentType<{}> = styled.button`
  color: red;
`;
