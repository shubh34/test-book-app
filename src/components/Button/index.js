import React from 'react';
import { StyledButton } from './styles';

const Button = ({ id, name, onClick = () => {} }) => {
  return (
    <StyledButton data-testid={id} id={id} name={name} onClick={onClick}>
      {name}
    </StyledButton>
  );
};

export default Button;
