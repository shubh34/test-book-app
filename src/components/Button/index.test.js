import userEvent from '@testing-library/user-event';
import Button from '.';
import { setUpComponent } from '../../../test/utils';

const setProps = (setUpProps = {}) => ({
  id: 'button-id',
  name: 'Button name',
  onClick: jest.fn(),
  ...setUpProps
});

describe('Button', () => {
  it('renders without crashing', () => {
    const { container } = setUpComponent(Button, setProps());
    expect(container).toMatchSnapshot();
  });
  it('should handle onClick', () => {
    const { getByTestId, props } = setUpComponent(Button, setProps());
    userEvent.click(getByTestId(props.id));
    expect(props.onClick).toBeCalled();
  });
});
