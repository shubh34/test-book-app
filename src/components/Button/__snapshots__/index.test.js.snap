// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<div>
  <button
    class="sc-bdfBwQ kQBUXH"
    data-testid="button-id"
    id="button-id"
    name="Button name"
  >
    Button name
  </button>
</div>
`;
